src/box/sql/build.c

```diff
@@ -736,26 +736,32 @@ sql_create_check_contraint(struct Parse *parser, bool is_field_ck)
 void
 sqlAddCollateType(Parse * pParse, Token * pToken)
 {
+	char *coll_name = sql_name_from_token(pToken);
```

Comment:
Can we use `sql_tt_name_from_token` here? I see the main blocker within `sql_get_coll_seq` that checks `coll_name` against `NULL`. Is this the only reason for not using the static buffer? I believe, we can omit this change in scope of this PR, but it would be nice to make a research/refactoring for such places in SQL code.

Reply:
We cannot use sql_tt_name_from_token() because static buffer is limited and it is possible that name length will be more than static buffer size. It does not matter for an errors and comments, but it is important in all other places. Though, there will be added sql_parser_name(), which will allow to use region from parser for temporary allocations, but not in this PR.

Reply:
Got it, thanks for clarification!